### ISSUE TITLE
Document upgrade and downgrade processes

### ISSUE THREAD

**Issue by zamazan4ik** [dragonflydb/dragonfly#226]
Hi.

After reading the documentation it is not clear for me, how upgrade and downgrade processes should be performed over the database. Could you please describe somewhere in the documentation:

- How to upgrade the database to the newer version (hopefully with no downtime)
- How to downgrade the database to the older version (hopefully with no downtime)
- Maybe some compatibility policies and notes between versions

Thanks in advance!

**Comment by romange** [comment]
yes, we need to document this. Currently, Dragonfly does not support primary/secondary replication. therefore it's not possible to upgrade without downtime.

with downtime - it's similar to Redis OSS: you produce an rdb snapshot on the running server with "SAVE" command. 
Then you use `dbfilename` flag to load the snapshot on a new server.

**Comment by zamazan4ik** [comment]
@romange what about downgrade?

**Comment by romange** [comment]
I understand what you are asking. We launched 3 months ago and we do not guarantee that future DF versions will necessarily be backward compatible with earlier ones. Currently we are at a stage where innovation precedes compatibility. 

Having said that, unless you used streams, your rdb snapshots should be compatible with all versions of dragonfly.
